sklearn/calibration.py

```diff
@@ -753,3 +755,209 @@ def calibration_curve(y_true, y_prob, *, normalize=False, n_bins=5,
     prob_pred = bin_sums[nonzero] / bin_total[nonzero]
 
     return prob_true, prob_pred
+
+
+class CalibrationDisplay:
+    """Calibration visualization.
+
+    It is recommend to use :func:`~sklearn.calibration.plot_calibration_curve`
+    to create a visualizer. All parameters are stored as attributes.
+
+    Read more in the :ref:`User Guide <calibration>`.
+
+    Parameters
+    -----------
+    prob_true : ndarray
+        The proportion of samples whose class is the positive class (fraction
+        of positives), in each bin.
+
+    prob_pred : ndarray
+        The mean predicted probability in each bin.
+
+    y_prob : ndarray of shape (n_samples,)
+        Probability estimates for the positive class, for each sample.
+
+    estimator_name : str, default=None
+        Name of estimator. If `None`, then the estimator name is not shown.
+
+    Attributes
+    ----------
+    line_ : matplotlib Artist
+        Calibration curve.
+
+    ax_ : matplotlib Axes
+        Axes with calibration curve.
+
+    figure_ : matplotlib Figure
+        Figure containing the curve.
+    """
+    def __init__(self, prob_true, prob_pred, y_prob, *, estimator_name=None):
+        self.prob_true = prob_true
+        self.prob_pred = prob_pred
+        self.y_prob = y_prob
+        self.estimator_name = estimator_name
+
+    def plot(self, ax=None, *, name=None, ref_line=True, **kwargs):
+        """Plot visualization.
+
+        Extra keyword arguments will be passed to matplotlib's `plot`.
+
+        Parameters
+        ----------
+        ax : Matplotlib Axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        name : str, default=None
+            Name of calibration curve for labeling. If `None`, use the
+            name of the estimator.
```

Comment:
@thomasjpfan can you remind me why we allow to pass both `estimator_name` and `name` here? This was done in `plot_precision_recall` too

Reply:
I think they are essentially the same as we use `name` to set `estimator_name`:
```Python
    name = name if name is not None else estimator.__class__.__name__
    viz = CalibrationDisplay(
        prob_true=prob_true, prob_pred=prob_pred, y_prob=y_prob,
        estimator_name=name
    )
```
I guess they could be called the same thing? The user would generally only use `name` from `plot_calibration_curve`.